spec/models/chargeback_configured_system_spec.rb

```diff
@@ -0,0 +1,111 @@
+RSpec.describe ChargebackConfiguredSystem do
+  include Spec::Support::ChargebackHelper
+
+  let(:admin) { FactoryBot.create(:user_admin) }
+  let(:base_options) do
+    {:interval_size       => 2,
+     :end_interval_offset => 0,
+     :tag                 => '/managed/environment/prod',
+     :ext_options         => {:tz => 'UTC'},
+     :userid              => admin.userid}
+  end
+  let(:hourly_rate)               { 0.01 }
+  let(:count_hourly_rate)         { 1.00 }
+  let(:cpu_count)                 { 1.0 }
+  let(:memory_available)          { 1000.0 }
+  let(:starting_date) { Time.parse('2012-09-01 23:59:59Z').utc }
+  let(:ts) { starting_date.in_time_zone(Metric::Helper.get_time_zone(base_options[:ext_options])) }
+  let(:report_run_time) { month_end }
+  let(:month_beginning) { ts.beginning_of_month.utc }
+  let(:month_end) { ts.end_of_month.utc }
+  let(:hours_in_month) { Time.days_in_month(month_beginning.month, month_beginning.year) * 24 }
+  let(:ems) { FactoryBot.create(:ems_vmware) }
+
+  let(:hourly_variable_tier_rate)       { {:variable_rate => hourly_rate.to_s} }
+  let(:count_hourly_variable_tier_rate) { {:variable_rate => count_hourly_rate.to_s} }
+  let(:fixed_compute_tier_rate)         { {:fixed_rate => hourly_rate.to_s} }
+
+  let(:detail_params) do
+    {
+      :chargeback_rate_detail_cpu_allocated      => {:tiers => [count_hourly_variable_tier_rate]},
+      :chargeback_rate_detail_memory_allocated   => {:tiers => [hourly_variable_tier_rate]},
+      :chargeback_rate_detail_fixed_compute_cost => {:tiers => [fixed_compute_tier_rate]}
+    }
+  end
+
+  let!(:chargeback_rate) do
+    FactoryBot.create(:chargeback_rate, :detail_params => detail_params)
+  end
+
+  before do
+    MiqRegion.seed
+    ChargebackRateDetailMeasure.seed
+    ChargeableField.seed
+    ManageIQ::Showback::InputMeasure.seed
```

Comment:
this is not needed.

Reply:
I am not able to get the test to pass without these seeding. For example, when i comment out ManageIQ::Showback::InputMeasure.seed, i am getting and similarly for ChargeableField.seed

NoMethodError:
  undefined method `adjust' for nil:NilClass
  # ./app/models/chargeable_field.rb:62:in `adjustment_to'
  # ./app/models/chargeback_rate_detail.rb:171:in `rate_adjustment'